### PR TITLE
SLE: Disable pids controller limit under user instances

### PR DESCRIPTION
**Pull request by Werkov**
We disable tasks limits but pids controller remains enabled because of DefaultTasksAccounting=yes. Without limits, accounting is only informational and it pulls in the pids controller as the only one. Clean environments with disabled controller delegation even from this one. (Although pids controller does not incur measurable overhead.)

References: jsc#PED-2276

### CODE REVIEW

**Comment by fbuihuu** [comment]
Given that the pids controller has always been enabled on all products (where it's supported) I wouldn't bother change that for the only sake of consistency and avoid the risk of someone complaining about the (feature) regression. I mean if it doesn't incur performance overhead, why not just simply accepting that pids controller is the only controller enabled by default ?

**Comment by fbuihuu** [comment]
and sorry for the delay...

**Comment by Werkov** [comment]
I see the point of not changing in the past/existing deployments. Let's wait how #3 turns out wrt SP-ecification.

**Comment by Werkov** [comment]
Fair enough, keeping pid controller system-wide enablement as it has always been. (It's worthy for me to have this recorded as explanation why `pids` controller remains enabled.)

Thus I drop commit [SLE: Disable pids controller via TasksAccounting](https://github.com/openSUSE/systemd-default-settings/pull/2/commits/4135a73e98b74aae78d601c6347cebc1ea2caad7) but I kept the other one that is a new thing that applies to user instances (again, it only has effect on unified setups because hybrid have no pids controller delegation inside user instance subtrees).

**Comment by Werkov** [comment]
> Thus I drop commit [SLE: Disable pids controller via TasksAccounting](https://github.com/openSUSE/systemd-default-settings/pull/2/commits/4135a73e98b74aae78d601c6347cebc1ea2caad7) but I kept the other one that is a new thing that applies to user instances

...and updated PR title accordingly.

File: usr/lib/systemd/user.conf.d/__25-defaults-SLE.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#
+# Do not edit this file, it will be overwritten on update.
+#
+# If you need to override one of the options, please create your own
+# drop-in in /etc/systemd with a higher priority (preferably greater
+# than 70).
+
+[Manager]
+# Disable default TasksMax limit (jsc#SLE-10123)
+DefaultTasksMax=infinity
```
Comment (fbuihuu) [inline usr/lib/systemd/user.conf.d/__25-defaults-SLE.conf:11]: Isn't it redundant with usr/lib/systemd/system/user-.slice.d/25-defaults-SLE.conf ?

Comment (Werkov) [reply usr/lib/systemd/user.conf.d/__25-defaults-SLE.conf:11]: * `user-.slice.d/25-defaults-SLE.conf` applies to each `user-$uid.slice`
  * path `user.slice/user-$uid.slice`
* this PR's config affects what is done to (user) units under `user@$uid.service` subtree
  * path `user.slice/user-$uid.slice/user@$uid.service/foo.unit`
  * also this is user writable thus a "voluntary" limit

So, when the former gives free-run, the latter could still limit underlying (user) unit (with upstream default).

Comment (fbuihuu) [reply usr/lib/systemd/user.conf.d/__25-defaults-SLE.conf:11]: yeah... I mixed things up :P

Could you extend the comment so it clearly states that the limit is removed for all units spawned by the user instances ?

Thanks.

**Comment by Werkov** [comment]
Force-pushed the in-file comment update.

**Comment by fbuihuu** [comment]
Thanks.

**Pull request merged**